lib/datadog/tracing/contrib/sinatra/integration.rb

```diff
@@ -31,6 +31,14 @@ def new_configuration
             Configuration::Settings.new
           end
 
+          def subconfigure(tracing, sinatra)
+            tracing.instrument(
+              :rack,
+              service_name: sinatra.service_name,
```

Comment:
I wonder if we should deprecate setting the rack `service_name` from the sinatra service name for 2.0?

E.g. instead suggest customers configure a top-level service name instead, or if they really want to, they can configure a service name in rack directly.

Reply:
It's not so much about passing a manually configured service name as much as when that `sinatra.service_name` is defaulted and inferred e.g from the app class or something.

Same for Rails where it's the application name as a default.

Tangentially there may be a conundrum to resolve here in such default cases: ordering may matter when there's both Sinatra and Rails in the same app, as both would set their service name and one would win. But that's a problem for another day.